application/src/main/java/org/togetherjava/tjbot/commands/ReloadCommand.java

```diff
@@ -0,0 +1,116 @@
+package org.togetherjava.tjbot.commands;
+
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.events.interaction.ButtonClickEvent;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import net.dv8tion.jda.api.interactions.components.Button;
+import net.dv8tion.jda.api.interactions.components.ButtonStyle;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.requests.restaction.CommandListUpdateAction;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReloadCommand implements Command {
+    private final CommandHandler commandHandler;
+
+    public ReloadCommand(CommandHandler commandHandler) {
+        this.commandHandler = commandHandler;
+    }
+
+    @Override
+    public @NotNull String getCommandName() {
+        return "reload";
+    }
+
+    @Override
+    public @NotNull String getDescription() {
+        return "Uploads all existing slash-commands to Discord so they're fully up-to-date.";
+    }
+
+    @Override
+    public boolean isGuildOnly() {
+        return true;
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandEvent event) {
+        Member member = event.getMember();
+
+        if (member.hasPermission(Permission.MANAGE_SERVER)) {
+            event.reply(
+                    "You sure? You can only reload commands a few times each day, so don't overdo this.")
+                .addActionRow(
+                        Button.of(ButtonStyle.SUCCESS, generateComponentId(member.getId()),
+                                "Affirmative!"),
+                        Button.of(ButtonStyle.DANGER, generateComponentId(member.getId()),
+                                "Wrong command! Sorry"))
+                .queue();
+        } else {
+            event.reply("You need the MANAGE_SERVER permission to use this command!")
+                .setEphemeral(true)
+                .queue();
+        }
+    }
+
+    @Override
+    public void onButtonClick(ButtonClickEvent event, List<String> idArgs) {
+        Member member = event.getMember();
+
+        if (member.getId().equals(idArgs.get(0))) {
+            switch (event.getButton().getStyle()) {
+                case DANGER -> {
+                    event.reply("Next time use the right command!").queue();
+                    event.getMessage().editMessageComponents().queue();
+                }
+
+                case SUCCESS -> {
+                    event.deferReply().queue();
+                    List<Command> commands = commandHandler.getCommandList();
+                    List<RestAction<List<net.dv8tion.jda.api.interactions.commands.Command>>> restActions =
+                            new ArrayList<>();
+
+                    // * adds all the global commands
+                    CommandListUpdateAction globalCommandsUpdateAction =
```

Comment:
This is with regards commentary around extracting methods from large lambdas eg.
```java
private final RestAction<List<net.dv8tion.jda.api.interactions.commands.Command>> restActionsForGlobalCommandUpdate(Event, List<Command>)
```